src/mongo/client/dbclientinterface.h

```diff
@@ -897,7 +897,13 @@ namespace mongo {
 
            returns true if successful
         */
-        bool copyDatabase(const std::string &fromdb, const std::string &todb, const std::string &fromhost = "", BSONObj *info = 0);
+        bool copyDatabase(const std::string &fromdb,
+                          const std::string &todb,
+                          const std::string &fromhost = "",
+                          const std::string &mechanism = "DEFAULT",
```

Comment:
You might consider providing SCRAM-SHA-1 or MONGODB-CR as the default to simplify the implementation.